fields.go

```diff
@@ -116,3 +100,22 @@ func NamedError(key string, err error) Field {
 	}
 	return zap.NamedError(key, &encoders.ErrorEncoder{Source: err})
 }
+
+// ptrValueTypes lists all acceptable pointer types for creating fields.
+type ptrValueTypes interface {
+	~string | ~int | ~int32 | ~int64 | ~uint | ~uint32 | ~uint64 | ~float32 | ~float64 | ~bool
+}
+
+// Ptr safely dereferences a value for use in a log field, rendering a zero value if the
+// value is nil. For example:
+//
+//   var value *string
+//   log.String("key", log.Ptr(value)) // {"key":"key","value":""}
+//
+func Ptr[T ptrValueTypes](value *T) T {
+	if value == nil {
+		var zero T
+		return zero
```

Comment:
I'm thinking about any unexpected side effects of this implementation – it feels slightly odd to be logging something else than what was passed in. I guess it shouldn't have to be an issue since it's just for the purpose of having a readable equivalent that doesn't raise an error, but wanted to point it out regardless.